test/e2e/sidecar_test.go

```diff
@@ -21,48 +24,80 @@ import (
 	"github.com/jaegertracing/jaeger-operator/pkg/inject"
 )
 
-// Sidecar runs a test with the agent as sidecar
-func Sidecar(t *testing.T) {
-	ctx := prepare(t)
-	defer ctx.Cleanup()
+type SidecarTestSuite struct {
+	suite.Suite
+}
 
-	if err := sidecarTest(t, framework.Global, ctx); err != nil {
-		t.Fatal(err)
-	}
+func(suite *SidecarTestSuite) SetupSuite() {
+	t = suite.T()
+	ctx = prepare(t)
+	fw = framework.Global
+	namespace, _ = ctx.GetNamespace()
+	require.NotNil(t, namespace, "GetNamespace failed")
+
+	addToFrameworkSchemeForSmokeTests(t)
+}
+
+func (suite *SidecarTestSuite) TearDownSuite() {
+	ctx.Cleanup()
 }
 
-func sidecarTest(t *testing.T, f *framework.Framework, ctx *framework.TestCtx) error {
+func TestSidecarSuite(t *testing.T) {
+	suite.Run(t, new(SidecarTestSuite))
+}
+
+// Sidecar runs a test with the agent as sidecar
+func (suite *SidecarTestSuite) TestSidecar() {
 	cleanupOptions := &framework.CleanupOptions{TestContext: ctx, Timeout: timeout, RetryInterval: retryInterval}
-	namespace, err := ctx.GetNamespace()
-	if err != nil {
-		return fmt.Errorf("could not get namespace: %v", err)
-	}
 
-	j := &v1.Jaeger{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "Jaeger",
-			APIVersion: "jaegertracing.io/v1",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "agent-as-sidecar",
-			Namespace: namespace,
-		},
-		Spec: v1.JaegerSpec{
-			Strategy: "allInOne",
-			AllInOne: v1.JaegerAllInOneSpec{},
-			Agent: v1.JaegerAgentSpec{
-				Options: v1.NewOptions(map[string]interface{}{
-					"log-level": "debug",
-				}),
-			},
-		},
-	}
+	j := getJaegerDefinition(namespace)
```

Comment:
Should be named something like `getJaegerAgentAsSidecarDefinition` just to distinguish it from other methods also returning CRs.